index.js

```diff
@@ -58,10 +58,21 @@ class TemplateCompiler extends Filter {
   }
 
   processString(string, relativePath) {
-    return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
-      contents: string,
-      moduleName: relativePath
-    }) + ';';
+    try {
+      return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
+        contents: string,
+        moduleName: relativePath
+      }) + ';';
+    } catch(e) {
+      // Create a new error, augment it with location and type information
+      // and re-throw.
+      let error = new Error(e);
```

Comment:
Seems _slightly_ odd.  Why don't we just add the props we need/want to the existing error?  I think as it stands we loose the original stack info?

Also, I believe that broccoli-persistent-filter also wraps in a try / catch so that it can add the file name info...